File: grocart_backend/src/main/java/com/example/grocart_backend/model/InternetItem.java
package com.example.grocart_backend.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.Data; // If using Lombok, otherwise generate getters/setters

@Entity
@Data
public class InternetItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // Matches your Android SerialName AND ViewModel logic
    @JsonProperty("stringResourceId")
    private String itemName;

    @JsonProperty("itemCategoryId")
    private String itemCategory;

    @JsonProperty("itemQuantity")
    private String itemQuantity;

    // CRITICAL: Change name to itemPrice to avoid any confusion with integer mapping
    @JsonProperty("item_price")
    private Integer itemPrice; // Using Integer instead of int is safer for JSON parsing

    @JsonProperty("imageResourceId")
    private String imageUrl;
}

File: grocart_frontend/app/src/main/java/com/grocart/first/data/InternetItem.kt
package com.grocart.first.data

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
// ✅ SERIALIZABLE CLASSES FOR ITEMS AND ORDERS
@Serializable
data class InternetItem(
    @SerialName("stringResourceId") val itemName: String = "",
    @SerialName("itemCategoryId") val itemCategory: String = "",
    @SerialName("itemQuantity") val itemQuantity: String = "",
    @SerialName("item_price") val itemPrice: Int = 0,
    @SerialName("imageResourceId") val imageUrl: String = ""

)